### DISCUSSION TITLE
Include a column's max length in code generated from DB introspection

### DISCUSSION THREAD

**Discussion by blaas** [diesel-rs/diesel#3551] [Ideas]
Hello everyone!

**Situation:**
My use case is the following: I have an API, and I perform validation of incoming requests. A lot of text fields directly map to a column in my database. In this case, I want to keep the API validation check consistent with the column's max length in database.

**Current solution:**
Currently, I need to define the max length once in my validation code, and once in the DB schema. This is duplicating the information, and causes bugs when the max length is changed in one place but not the other (which I've seen happen in the past).

**Solution if Diesel supported it:**
The validation code could directly use the Diesel-defined constant corresponding to the max length of the field, guaranteeing consistency.

**Interface proposition:**
Given a table `recipes` with a column `title` of type `VARCHAR(128)`:
```rust
assert_eq!(schema::recipes::title::MAX_LENGTH, 128);
```
No constant would be defined on columns for which it isn't relevant.

**Disadvantages:**
- None that I can think of

I don't know much about Diesel's internals, but I feel like it would be a decent small feature to add to the library. I hope you'll consider it!

**Comment by Ten0** [comment]
Looks like auto mention doesn't work on discussions :o 
https://github.com/diesel-rs/diesel/pull/3552
